File: swift/Sources/JchuComponentsExtensions/Foundation/BooleanExtensions.swift
import Foundation

public extension Optional where Wrapped == Bool {
    /// Returns the wrapped value or a supplied fallback.
    func orFalse(defaultValue: Bool = false) -> Bool {
        self ?? defaultValue
    }

    /// Whether the optional contains `true`.
    var isTrue: Bool {
        self == true
    }
}

File: jchucomponents-foundation/src/commonMain/kotlin/com/jeluchu/jchucomponents/foundation/extensions/BooleanExtensions.kt
package com.jeluchu.jchucomponents.foundation.extensions

/**
 * Returns this value, or [defaultValue] when it is `null`.
 */
fun Boolean?.orFalse(defaultValue: Boolean = false): Boolean = this ?: defaultValue

/**
 * Returns `true` only when this nullable value is explicitly `true`.
 */
fun Boolean?.isTrue(): Boolean = this == true
